mlx/c/metal.cpp

```diff
@@ -3,6 +3,8 @@
 /* This file is auto-generated. Do not edit manually. */
 /*                                                    */
 
+#include <cstring>
```

Comment:
see comment above

Reply:
Noted, i.e. moving it to `mlx/c/private/mlx.h`.

Reply:
I tried replacing it with `<string.h>`, but it failed - again complaining that there is no `strncpy`: https://github.com/stemann/mlx-c/actions/runs/11923235104

Using <cstring> (now in `mlx/c/private/mlx.h`) fixes it: https://github.com/stemann/mlx-c/actions/runs/11923278948